steps/24_dataset_object/step.md

```diff
@@ -0,0 +1,349 @@
+# Dataset object
+
+Contributors: ["felipeangelimvieira", "fkiraly"]
+
+## Introduction
+
+This STEP is an enhacement proposal concerning the creation of Dataset objects.
+
+For preliminary discussions of the proposal presented here, see issues and PRs: 
+
+* sktime/sktime#4332
+* sktime/sktime#4333
+* sktime/sktime#7398
+
+## Contents
+
+1. [Introduction](#introduction)
+2. [Problem Statement](#problem-statement)
+3. [Description of Proposed Solution](#description-of-proposed-solution)  
+   - [__init__ and load](#__init__-and-load)  
+   - [Train and Test Splits](#train-and-test-splits)  
+   - [Base Classes](#base-classes)  
+   - [Tags](#tags)  
+     - [Base Dataset Tags](#base-dataset-tags)  
+     - [Classification Tags](#classification-tags)  
+     - [Regression Tags](#regression-tags)  
+     - [Forecasting Tags](#forecasting-tags)  
+4. [Motivation](#motivation)
+5. [Discussion and Comparison of Alternative Solutions](#discussion-and-comparison-of-alternative-solutions)
+6. [Detailed Description of Design and Implementation](#detailed-description-of-design-and-implementation-of-proposed-solution)  
+
+## Problem statement
+
+Currently, datasets in sktime are loaded by the usage of functions, and having a dataset object would provide many benefits for documentation purposes. For example, in the case of estimators, there's an estimator overview page that allows users to filter and find estimators with certain characteristics for classification, forecasting and regression.
+Having dataset objects would allow the retrieval of problems with characteristics that fit the user needs by the usage of the tag system.
+
+In addition to this, an unified interface that also identifies the characteristics of each dataset can be useful for benchmarking purposes. If a researcher is investigating in which circunstances a model performs well, the metadata (tags) of the datasets could be useful to analize and understand these factors.
+
+The dataset object interface could also be used for other scenarious beyond timeseries forecasting, classification and regression. For example, the forecasting dataset object could be used for Marketing Mix Modeling datasets (see felipeangelimvieira/prophetverse#101)
+
+## Description of proposed solution
+
+### `__init__` and `load`
+
+Dataset objects should have all the parameters that identify the content of it defined in `__init__`, similar to what hyperparameters represent for estimators. The `load(*args)` method is used to obtain the sets. This method receives strings as positional args, such as `load("X", "y", "X_train", "y_train", "X_test", "y_test")` and returns the respective sets. Hence, users can define which set should be retrieved by passing its name to `load`. 
```

Comment:
should this handle type specifications, or not? The loaders have a `return_mtype` arg, but the returns could be of different mtypes or even scitypes (e.g., classification has `Panel` and `Table`)

Maybe for now we can focus on `pd.DataFrame` based types and extned late.r